src/api/Nhs.Appointments.Persistance/UserStore.cs

```diff
@@ -33,7 +33,7 @@ public async Task<IEnumerable<RoleAssignment>> GetUserRoleAssignments(string use
     /// <param name="userId"></param>
     /// <param name="scope"></param>
     /// <param name="roleAssignments"></param>
-    /// <returns>The original role assignments prior to completion of the update operation</returns>
+    /// <returns>The original role assignments for the scope prior to completion of the update operation</returns>
```

Comment:
I know this isn't a change you've introduced, but this really jarred me when I first read it. I would expect this to either return nothing at all, or the user's roles as they NOW are, after the update has been applied. 
Looking at the code, it seems we're relying on this method giving us old info to perform the following logic: 
<img width="580" alt="image" src="https://github.com/user-attachments/assets/db309829-097c-41b4-a514-b4f20b5c40f1" />

Reply:
Part of me wonders if it'd be better to keep this method uncomplicated, and filter by scope in the block above. That keeps a better separation of concerns. Equally though, I don't think we should be using this method to derive the "old" roles anyway so... 🤷